### DISCUSSION TITLE
Docker setup Issue on arm64 architecture

### DISCUSSION THREAD

**Discussion by a6ar55** [learningequality/kolibri#11353] [Q&A] [answered]
While setting up the docker for kolibri i got some issues while executing "make docker-build-base"  later figured out that it was due to the difference in the architecture ,the current one works for only amd64 for some packages(node js and yarn) while mine is arm64. 

should i raise an issue or pull request (i made a working base.dockerfile for arm 64) ?

<img width="701" alt="Screenshot 2023-10-05 at 3 46 40 PM" src="https://github.com/learningequality/kolibri/assets/117556787/ce36c5f1-6a43-4dbd-8115-7ac5f2a30123">

**Comment by bjester** [comment, marked as answer]
Hi @a6ar55,

Thank you for your interest in contributing.

I think the appropriate fix is to update the dockerfile to instead parameterize `amd64` with the result of the following command:
```bash
dpkg --print-architecture
```

That should allow it to work for both `amd64` and `arm64` host architectures

**Reply by a6ar55** [reply]
thank you @bjester,

I will proceed with making this change and submitting a pull request. Your assistance is invaluable, and I'm eager in contributing and improving the project.
thanks again.

**Comment by a6ar55** [comment]
Hey @bjester  I updated the "# install nodejs and yarn" and it successfully downloads the packages according to the architecture. 
<img width="568" alt="Screenshot 2023-10-10 at 6 13 24 PM" src="https://github.com/learningequality/kolibri/assets/117556787/8d283fef-ca4e-4495-84e9-3091651f0d9d">
But now I am getting issues with the installation of sqlalchemy which i think is due to the compatibility issue with the python version

I'm reaching out to get your suggestions on resolving this issue. It seems that SQLAlchemy 1.4.49 may not be compatible with Python 2.7.18. I'd appreciate any insights or recommendations you might have to make this work within the Python 2.7 environment itself.

<img width="811" alt="Screenshot 2023-10-10 at 6 14 02 PM" src="https://github.com/learningequality/kolibri/assets/117556787/a259dde8-1433-4fab-bf04-021077c922c7">

**Reply by bjester** [reply]
We have a few different dockerfiles, but the `base.dockerfile` does not seem to be a foundation for any build tooling (e.g. `build_whl.dockerfile`), so I think we can update it to use python3. Hopefully that will help resolve the issue.

**Reply by a6ar55** [reply]
okey ,thanks.

**Reply by DXCanas** [reply]
Just chiming in:

`base.dockerfile` is actually used in the `docker-build-base` Make recipe. After being built, it's tagged as `learningequality/kolibribase`. 

`learningequality/kolibribase` is the tag used for [`dev.dockerfile`](https://github.com/learningequality/kolibri/blob/ef9a11b0ef4e9d5f5e968eef37fe7934d0a1e0fc/docker/dev.dockerfile#L4) and [`demoserver.dockerfile`](https://github.com/learningequality/kolibri/blob/ef9a11b0ef4e9d5f5e968eef37fe7934d0a1e0fc/docker/demoserver.dockerfile#L4); [search results here.](https://github.com/search?q=org%3Alearningequality+learningequality%2Fkolibribase&type=code)

`dev.dockerfile` is actually what's used for the Make recipe for `docker-devserver`:
https://github.com/learningequality/kolibri/blob/ef9a11b0ef4e9d5f5e968eef37fe7934d0a1e0fc/Makefile#L307-L320

So @bjester is right that it's not used for build tooling (it seems), but it might be used for the docker dev workflow (which is I guess why you're working on this at all!). The only issues I can potentially foresee are attempting to build a complete Kolibri WHL while in this dev mode; but as it's a Dockerized instance, I don't think that's a common use case.
